tests/Tracing/TransactionContextTest.php

```diff
@@ -80,4 +85,98 @@ public function fromSentryTraceDataProvider(): iterable
             null,
         ];
     }
+
+    /**
+     * @dataProvider fromHeadersDataProvider
+     */
+    public function testFromHeaders(string $sentryTraceHeader, string $baggageHeader, ?SpanId $expectedSpanId, ?TraceId $expectedTraceId, ?bool $expectedParentSampled, ?string $expectedDynamicSamplingContextClass, ?bool $expectedDynamicSamplingContextFrozen)
+    {
+        $spanContext = TransactionContext::fromHeaders($sentryTraceHeader, $baggageHeader);
+
+        if (null !== $expectedSpanId) {
```

Comment:
Calling `assertEquals()` should be enough, no matter what's the expected value is, because it covers also the case where `$expectedSpanId === null`